src/dynamic_honey_badger/mod.rs

```diff
@@ -111,20 +109,86 @@ pub enum Message<N: Rand> {
     SignedVote(SignedVote<N>),
 }
 
-impl<N: Rand> Message<N> {
+impl<N: Rand> MessageContent<N> {
     fn start_epoch(&self) -> u64 {
         match *self {
-            Message::HoneyBadger(epoch, _) => epoch,
-            Message::KeyGen(epoch, _, _) => epoch,
-            Message::SignedVote(ref signed_vote) => signed_vote.era(),
+            MessageContent::HoneyBadger(start_epoch, ..) => start_epoch,
+            MessageContent::KeyGen(start_epoch, ..) => start_epoch,
+            MessageContent::SignedVote(ref signed_vote) => signed_vote.era(),
```

Comment:
Ah, I hadn't realized that we were already inconsistent with the names: We should either rename `start_epoch` to `era` everywhere or vice versa. (But in a separate PR, I guess.)